src/app/views/datasets-new/create/CreateDatasetController.jsx

```diff
@@ -0,0 +1,161 @@
+            })
+            .catch(error => {
+                log.event("Error getting outputs with no created dataset", log.error(error));
+                const notification = {
+                    type: "warning",
+                    message: "An error occurred when trying to get available datatsets. Try refreshing the page",
+                    isDismissable: true
+                };
+                notifications.add(notification);
+                console.error("Error getting outputs with no created dataset:\n", error);
+                this.setState({ isFetchingRecipesAndDatasets: false });
+            });
+    };
+
+    // A recipe can output many datasets, so loop through and get
+    // all possible outputs that can be created from a single recipe
+    getAllOutputsFromRecipes = recipes => {
+        const allOutputs = [];
+        recipes.items.forEach(recipe => {
+            if (recipe.output_instances.length > 1) {
+                recipe.output_instances.forEach(output => {
+                    allOutputs.push(output);
+                });
+                return;
+            }
+            allOutputs.push(recipe.output_instances[0]);
+        });
+        return allOutputs;
+    };
+
+    getOutputsWithoutExistingDataset = (outputs, datasets) => {
+        return outputs.filter(output => {
+            const doesMatchRecipe = datasets.items.some(datasetItem => {
+                return output.dataset_id === datasetItem.id;
+            });
+            return !doesMatchRecipe;
+        });
+    };
+
+    mapOutputsToState = outputs => {
+        return outputs.map(output => {
+            return {
+                title: output.title,
+                id: output.dataset_id,
+                url: this.props.location.pathname + "/" + output.dataset_id
+            };
+        });
+    };
+
+    handleSearchInput = event => {
+        const searchTerm = event.target.value.toLowerCase();
+        const filteredOutputs = this.state.outputs.filter(recipe => {
+            return recipe.title.toLowerCase().search(searchTerm) !== -1;
+        });
+        this.setState({
+            filteredOutputs,
+            searchTerm
+        });
+    };
+
+    handleBackButton = () => {
+        const previousUrl = url.resolve("../");
+        this.props.dispatch(push(previousUrl));
+    };
+
+    render() {
+        return (
+            <div className="grid grid--justify-center">
+                <div className="grid__col-9">
+                    <div className="margin-top--2">
+                        &#9664;{" "}
+                        <button type="button" className="btn btn--link" onClick={this.handleBackButton}>
+                            Back
+                        </button>
+                    </div>
+                    <h1 className="margin-top--1 margin-bottom--1">Create new dataset</h1>
+                    <Input id="search-datasets" placeholder="Search by name or ID" onChange={this.handleSearchInput} />
```

Comment:
This is not quite accurate, you can only search by name, not by ID e.g. CPI would yeild no results but UK consumer price inflation would yield results.
```suggestion
                    <Input id="search-datasets" placeholder="Search by name" onChange={this.handleSearchInput} />
```